### ISSUE TITLE
Advanced Room and Portal usage Objects in the portal modes: ROAMING do not work

### ISSUE THREAD

**Issue by gitscher** [godotengine/godot#54908]
### Godot version

Godot_v3.4-stable_win64

### System information

Windows 10, i7-7700K, RTX 3080, GLES3

### Issue description

Game with a mix of outdoor and indoor environment.
Objects in the portal modes: ROAMING do not work, they are always displayed when there is at least one portal connecting from indoor to outdoor.

Play with a mix of outdoor and indoor environment without a portal connecting from indoor to outdoor and the objects with the portal mode: ROAMING in the indoor area are at the time of the Convert Rooms button press. They are not displayed in the outdoor area. The objects are displayed when you go to the indoor area. This is still correct but when the objects go from indoor to outdoor after the Convert Rooms button they are not updated.
These objects can only be seen in the indoor area but not in the outdoor area where they are located.

In games with normal Rooms and Portals system, objects in the Portal mode: ROAMING are not displayed outside the field of view. is correct.

### Steps to reproduce

1. Download project
2. Start project

First example: Convert Rooms button press

Second example: Delete portals that are connected to the outdoor area.
Then press the Convert Rooms button.
Move the red objects with the portal mode "ROMING" to the outdoor area.

third example: delete outdoor RoomGroup with room and also delete the portals with the names windows. Then press the Convert Rooms button.

### Minimal reproduction project

![editor_screenshot_2021-11-12T113203+0100](https://user-images.githubusercontent.com/94169968/141452985-8ce05c81-e388-4404-9e38-b1f930ffb75f.png)
)
![editor_screenshot_2021-11-12T113301+0100](https://user-images.githubusercontent.com/94169968/141453005-da282e47-189d-4339-8d02-c9db5f479978.png)
[Protal.zip](https://github.com/godotengine/godot/files/7527038/Protal.zip)

**Comment by Calinou** [comment]
cc @lawnjelly

**Comment by lawnjelly** [comment]
I've tracked this down, it's not too serious. What is happening is that roaming objects have an expansion margin to prevent them updating all the time (the BVH does this as well, completely different system, but a similar principle).

The expansion margin is automatically estimated based on the average bound size of the rooms in the room system. This metric works great with a standard indoor room system, but it turns out it goes pear shaped in your scene, because your outdoor room is huge, thus the expansion margin is huge, and the roaming objects are extending from the inner room to the outside (as this is done based on their expanded margin, not the tight AABB).

The metric at the moment isn't that good, it's something like this:
```
Find average room size (x/y/z). Find the longest axis of the average room size, multiply by 0.08.
```

In your case you had outdoors (2000x2000 or so) and a couple of small rooms. The average was large, so the expansion margin was coming out almost as big as the couple of rooms. This isn't your fault at all, this is an expected scenario with outdoor scenes, I just overlooked it.

I think the solution is either to tweak the metric or expose this to the user, probably exposing it to the user. We had a similar problem with the margins in the BVH. It's kind of difficult to make an automatic metric that works in all situations because users can build levels of any dimensions, and any combinations of dimensions, and their roaming objects could be any size, or move at any speed.

On the other hand the downside of exposing stuff like this to the user, is they can get confused. Generally we err on exposing as little as possible, but in this case it probably needs to be editable. I'll put in a fix for this. :+1:
